### PR TITLE
address review comments

### PR DESCRIPTION
**Pull request by ritazh**
- remove KMSv2Configuration
- separate EncryptResponse from EncryptedObject

Signed-off-by: Rita Zhang <rita.z.zhang@gmail.com>

  
- One-line PR description:

- Issue link:

- Other comments:

### CODE REVIEW

File: keps/sig-auth/3299-kms-v2-improvements/README.md [hunk 1]
```diff
@@ -126,27 +126,9 @@ proposal will be implemented, this is the place to discuss them.
...
- }
-+
-+type KMSv2Configuration struct {
-+    // name is the name of the KMSv2 plugin to be used.
-+    Name string `json:"name"`
-+    // endpoint is the gRPC server listening address, for example "unix:///var/run/kmsv2-provider.sock".
-+    Endpoint string `json:"endpoint"`
-+}
++    // apiversion of KeyManagementService
++    APIVersion string `json:"apiversion"`
```
Comment (aramase) [inline keps/sig-auth/3299-kms-v2-improvements/README.md]:
```suggestion
+    APIVersion string `json:"apiVersion"`
```

File: keps/sig-auth/3299-kms-v2-improvements/README.md [hunk 2]
```diff
@@ -225,7 +207,7 @@ The last byte represents the encoding style, with 0 meaning that the rest of the
 type EncryptedObject struct {
     TypeMeta `json:",inline" protobuf:"bytes,1,opt,name=typeMeta"`
 
-    EncryptResponse EncryptResponse `protobuf:"bytes,2,opt,name=encryptResponse"`
+    KeyId string `protobuf:"bytes,2,opt,name=keyId"`
```
Comment (aramase) [inline keps/sig-auth/3299-kms-v2-improvements/README.md]: In addition to keyId, we would store the entire annotations map here?

Comment (aramase) [inline keps/sig-auth/3299-kms-v2-improvements/README.md]: nit: should this be `KeyID` instead of `KeyId`?

Comment (ritazh) [reply keps/sig-auth/3299-kms-v2-improvements/README.md]: If we store the entire annotation here, how will it be used later? Anything extra we add here can reduce available etcd object size for the actual data right?

Comment (aramase) [reply keps/sig-auth/3299-kms-v2-improvements/README.md]: Annotations will contain the encrypted local KEK and additional details passed from the kms plugin to the kube-apiserver as part of `EncryptResponse`. We need to send it back to the kms plugin as part of `DecryptRequest` for it to use key hierarchy while decrypting and any other data that it needs from the annotations.

File: keps/sig-auth/3299-kms-v2-improvements/README.md [hunk 3]
```diff
@@ -225,11 +207,13 @@ The last byte represents the encoding style, with 0 meaning that the rest of the
...
     TypeMeta `json:",inline" protobuf:"bytes,1,opt,name=typeMeta"`
 
-    EncryptResponse EncryptResponse `protobuf:"bytes,2,opt,name=encryptResponse"`
+    KeyID string `protobuf:"bytes,2,opt,name=keyID"`
 
     PluginName string `protobuf:"bytes,3,opt,name=pluginName"`
 
     Ciphertext []byte `protobuf:"bytes,4,opt,name=ciphertext"`
+
+    Annotations map<string, bytes> `protobuf:"bytes,5,opt,name=annotations`
```
Comment (aramase) [inline keps/sig-auth/3299-kms-v2-improvements/README.md]:
```suggestion
    Annotations map[string][]byte `protobuf:"bytes,5,opt,name=annotations"`
```

File: keps/sig-auth/3299-kms-v2-improvements/README.md [hunk 4]
```diff
@@ -225,11 +207,13 @@ The last byte represents the encoding style, with 0 meaning that the rest of the
...
     TypeMeta `json:",inline" protobuf:"bytes,1,opt,name=typeMeta"`
 
-    EncryptResponse EncryptResponse `protobuf:"bytes,2,opt,name=encryptResponse"`
+    KeyID string `protobuf:"bytes,2,opt,name=keyID"`
 
     PluginName string `protobuf:"bytes,3,opt,name=pluginName"`
 
     Ciphertext []byte `protobuf:"bytes,4,opt,name=ciphertext"`
+
+    Annotations map[string][]byte `protobuf:"bytes,5,opt,name=annotations`
```
Comment (aramase) [inline keps/sig-auth/3299-kms-v2-improvements/README.md]:
```suggestion
    Annotations map[string][]byte `protobuf:"bytes,5,opt,name=annotations"`
```

**Review by aramase — APPROVED** [review]
lgtm

**Pull request merged**